utils/aeternity.js

```diff
@@ -64,61 +77,150 @@ class Aeternity {
 
       const eventsSchema = [
         { name: 'TipReceived', types: [SOPHIA_TYPES.address, SOPHIA_TYPES.int, SOPHIA_TYPES.string] },
+        { name: 'TipTokenReceived', types: [SOPHIA_TYPES.address, SOPHIA_TYPES.int, SOPHIA_TYPES.string, SOPHIA_TYPES.address] },
         { name: 'ReTipReceived', types: [SOPHIA_TYPES.address, SOPHIA_TYPES.int, SOPHIA_TYPES.string] },
+        { name: 'ReTipTokenReceived', types: [SOPHIA_TYPES.address, SOPHIA_TYPES.int, SOPHIA_TYPES.string, SOPHIA_TYPES.address] },
         { name: 'TipWithdrawn', types: [SOPHIA_TYPES.address, SOPHIA_TYPES.int, SOPHIA_TYPES.string] },
         { name: 'QueryOracle', types: [SOPHIA_TYPES.string, SOPHIA_TYPES.address] },
         { name: 'CheckPersistClaim', types: [SOPHIA_TYPES.string, SOPHIA_TYPES.address, SOPHIA_TYPES.int] },
+        { name: 'Transfer', types: [SOPHIA_TYPES.address, SOPHIA_TYPES.address, SOPHIA_TYPES.int] },
+        { name: 'Allowance', types: [SOPHIA_TYPES.address, SOPHIA_TYPES.address, SOPHIA_TYPES.int] },
       ];
 
       const decodedEvents = decodeEvents(tx.log, { schema: eventsSchema });
 
-      return decodedEvents.map(decodedEvent => ({
-        event: decodedEvent.name,
-        address: `ak_${decodedEvent.decoded[1]}`,
-        amount: decodedEvent.decoded[2] ? decodedEvent.decoded[2] : null,
-        url: decodedEvent.decoded[0],
-        caller: tx.tx.callerId,
-        nonce: tx.tx.nonce,
-        height: tx.height,
-        hash: tx.hash,
-        time: microBlock.time,
-      }));
+      return decodedEvents.map(decodedEvent => {
+        const event = {
+          event: decodedEvent.name,
+          caller: tx.tx.callerId,
+          nonce: tx.tx.nonce,
+          height: tx.height,
+          hash: tx.hash,
+          time: microBlock.time,
+          contract: tx.contractId,
+        };
+        switch (decodedEvent.name) {
+          case 'Transfer':
+            event.from = `ak_${decodedEvent.decoded[0]}`;
+            event.to = `ak_${decodedEvent.decoded[1]}`;
+            event.amount = decodedEvent.decoded[2] ? decodedEvent.decoded[2] : null;
```

Comment:
```suggestion
            event.amount = decodedEvent.decoded[2] || null;
```

Not that its not clear or something, but can this be simplified like so? 🤔

Reply:
I don't consider this simplification here, I prefer it to be explicit. Whats your opinion @kenodressel ?

Reply:
I agree with philipp, I like the explicit variant.